src/KeyedDistributions.jl

```diff
@@ -78,6 +85,17 @@ const MvNormalLike = Union{MvNormal, KeyedMvNormal}
 const KeyedGenericMvTDist = KeyedDistribution{Multivariate, Continuous, <:GenericMvTDist}
 const MvTLike = Union{GenericMvTDist, KeyedGenericMvTDist}
 
+# Use submat to preserve the covariance matrix PDMat type
+function (d::KeyedMvNormal)(keys...)::KeyedMvNormal
+    inds = map(x -> AxisKeys.findindex(x, axiskeys(d)[1]), vcat(keys[1]))
+    return KeyedDistribution(MvNormal(d.d.μ[inds], submat(d.d.Σ, inds)), vcat(keys...))
+end
+
+function (d::KeyedGenericMvTDist)(keys...)::KeyedGenericMvTDist
+    inds = map(x -> AxisKeys.findindex(x, axiskeys(d)[1]), vcat(keys[1]))
+    return KeyedDistribution(MvTDist(d.d.df, d.d.μ[inds], submat(d.d.Σ, inds)), vcat(keys...))
```

Comment:
Would using `collect` here do?
```suggestion
    return KeyedDistribution(MvTDist(d.d.df, d.d.μ[inds], submat(d.d.Σ, inds)), collect(keys))
```

Reply:
doesn't work with Symbol